core/src/main/scala/com/salesforce/op/stages/impl/preparators/SanityCheckerMetadata.scala

```diff
@@ -319,29 +323,45 @@ case class Correlations
       } else {
         corrType
       }
-    new Correlations(featuresIn ++ corr.featuresIn, values ++ corr.values, nanCorrs ++ corr.nanCorrs, corrName)
+    Correlations(featuresIn ++ corr.featuresIn, valuesWithLabel ++ corr.valuesWithLabel,
+      valuesWithFeatures ++ corr.valuesWithFeatures, corrName)
   }
 }
 
 case object SanityCheckerSummary {
 
   def flatten(checkers: Seq[SanityCheckerSummary]): SanityCheckerSummary = {
-    val correlationsWLabel: Correlations = checkers.map(_.correlationsWLabel).reduce(_ + _)
+    val correlations: Correlations = checkers.map(_.correlations).reduce(_ + _)
     val dropped: Seq[String] = checkers.flatMap(_.dropped)
     val featuresStatistics: SummaryStatistics = checkers.map(_.featuresStatistics).reduce(_ + _)
     val names: Seq[String] = checkers.flatMap(_.names)
     val categoricalStats: Array[CategoricalGroupStats] = checkers.flatMap(_.categoricalStats).toArray
-    new SanityCheckerSummary(correlationsWLabel, dropped, featuresStatistics, names, categoricalStats)
+    new SanityCheckerSummary(correlations, dropped, featuresStatistics, names, categoricalStats)
   }
 
   private def correlationsFromMetadata(meta: Metadata): Correlations = {
     val wrapped = meta.wrapped
-    Correlations(
-      featuresIn = wrapped.getArray[String](SanityCheckerNames.FeaturesIn).toSeq,
-      values = wrapped.getArray[Double](SanityCheckerNames.Values).toSeq,
-      nanCorrs = wrapped.getArray[String](SanityCheckerNames.CorrelationsWLabelIsNaN).toSeq,
-      corrType = CorrelationType.withNameInsensitive(wrapped.get[String](SanityCheckerNames.CorrelationType))
-    )
+    val features = wrapped.getArray[String](SanityCheckerNames.FeaturesIn).toSeq
+    if (wrapped.underlyingMap.keySet.contains("values")) { // old sanity checker meta
```

Comment:
nice backwards compatibility hack! maybe use "correlationsWithLabel" instead of "values", as its speaks more to the type of change that occurred?